src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,265 @@
+                tasks.add(task);
+
+                if(announce) {
+                    announceAddition(task);
+                }
+            }
+        }
+        return successful;
+    }
+
+    /**
+     * Adds an event task to the taskList.
+     *
+     * @param words the input broken down into individual words
+     * @param announce whether to announce the addition of the task
+     * @param isDone whether to set the task as done
+     * @return true if the addition was successful, false otherwise
+     */
+    public boolean addEvent(String[] words, boolean announce, boolean isDone) {
+        boolean successful = false;
+        int length = words.length;
+        int wordsIndex = 1;
+
+        if (length < 2) {
+            System.out.println("The name cannot be empty!");
+        } else {
+            StringBuilder nameBuilder = new StringBuilder();
+            while(wordsIndex < length) {
+                if (words[wordsIndex].equals("/from")) {
+                    wordsIndex++;
+                    break;
+                } else {
+                    nameBuilder.append(" ").append(words[wordsIndex]);
+                    wordsIndex++;
+                }
+            }
+            String name = nameBuilder.substring(1);
+
+            if (wordsIndex >= length) {
+                System.out.println("The start date cannot be empty!");
+            } else {
+                StringBuilder startDateBuilder = new StringBuilder();
+                while (wordsIndex < length) {
+                    if (words[wordsIndex].equals("/to")) {
+                        wordsIndex++;
+                        break;
+                    } else {
+                        startDateBuilder.append(" ").append(words[wordsIndex]);
+                        wordsIndex++;
+                    }
+                }
+                String startDate = startDateBuilder.substring(1);
+
+                if (wordsIndex >= length) {
+                    System.out.println("The end date cannot be empty!");
+                } else {
+                    StringBuilder endDateBuilder = new StringBuilder();
+                    while (wordsIndex < length) {
+                        endDateBuilder.append(" ").append(words[wordsIndex]);
+                        wordsIndex++;
+                    }
+                    String endDate = endDateBuilder.substring(1);
+                    successful = true;
+
+                    Task task = new Event(name, LocalDateTime.parse(startDate), LocalDateTime.parse(endDate), isDone);
+                    tasks.add(task);
+
+                    if(announce) {
+                        announceAddition(task);
+                    }
+                }
+            }
+        }
+        return successful;
+    }
+
+    /**
+     * Announces the addition of a task
+     *
+     * @param task the task being added
+     */
+    public void announceAddition(Task task) {
```

Comment:
Perhaps instead of annouce you can use print since it is more commonly used when we return a string